File: testprojects/python_main/pipelines/DONOT_OPEN_PYTHON/code/pythonorganization/donot/openme12/graph/src_jdbc_userandpass_test_table.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.libs import typed_lit
from pythonorganization.donot.openme12.config.ConfigStore import *
from pythonorganization.donot.openme12.udfs.UDFs import *

def src_jdbc_userandpass_test_table(spark: SparkSession) -> DataFrame:
    from pyspark.dbutils import DBUtils

    return spark.read\
        .format("jdbc")\
        .option("url", "jdbc:mysql://18.144.156.219:3306/test_database")\
        .option("user", DBUtils(spark).secrets.get(scope = "qasecrets_mysql", key = "username"))\
        .option("password", DBUtils(spark).secrets.get(scope = "qasecrets_mysql", key = "password"))\
        .option("dbtable", "test_table")\
        .option("pushDownPredicate", True)\
        .option("driver", "com.mysql.jdbc.Driver")\
        .load()

File: testprojects/scala_main/pipelines/SCALA_BASIC_20_MINS/code/src/main/scala/com/scala/main/job1/graph/src_jdbc_userandpass_test_table.scala
package com.scala.main.job1.graph

import io.prophecy.libs._
import com.scala.main.job1.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object src_jdbc_userandpass_test_table {

  def apply(context: Context): DataFrame = {
    var reader = context.spark.read
      .format("jdbc")
      .option("url",      "jdbc:mysql://18.144.156.219:3306/test_database")
      .option("user",     "test_user")
      .option("password", "admin")
      .option("dbtable",  "test_table")
    reader = reader
      .option("pushDownPredicate", true)
      .option("driver",            "com.mysql.jdbc.Driver")
    reader.load()
  }

}
